mongo/bulk_write.go

```diff
@@ -179,6 +180,13 @@ func (bw *bulkWrite) runInsert(ctx context.Context, batch bulkWriteBatch) (opera
 		Database(bw.collection.db.name).Collection(bw.collection.name).
 		Deployment(bw.collection.client.deployment).Crypt(bw.collection.client.cryptFLE).
 		ServerAPI(bw.collection.client.serverAPI)
+	if bw.comment != nil {
+		comment, err := transformValue(bw.collection.registry, bw.comment, false, "comment")
```

Comment:
```suggestion
		comment, err := transformValue(bw.collection.registry, bw.comment, true, "comment")
```

The third argument to `transformValue` specifies whether or not a map with multiple keys is allowed as the second argument (the `interface{}` to be transformed into a `bsoncore.Value`). 

I assume you took this logic from where we use `transformValue` on `hint` fields to transform them to `bsoncore.Value`s. The order of keys matters in a `hint` argument, as users could be referring to a [compound index](https://www.mongodb.com/docs/manual/core/index-compound/) that they'd like the query to make use of. `map`s in Go have an undefined order (`range`ing over them does not consistently produce the same order of keys). GODRIVER-1810 was an improvement to the driver to not allow users to pass in ambiguously ordered objects to arguments like `hint`. Thus, every transform of a `hint` in the driver passes `false` for `mapAllowed`.

`comment`, from what I gather, does not _need_ to be ordered in the same way, so I think we can pass `true` here and elsewhere for `mapAllowed`. What do others think?